packages/core/src/exports.ts

```diff
@@ -208,6 +209,15 @@ export function withScope<T>(
   return getCurrentHub().withScope(rest[0]);
 }
 
+/**
+ * TODO
+ */
+export function withIsolationScope<T>(callback: (isolationScope: Scope) => T): T {
+  return runWithAsyncContext(() => {
+    return callback(getIsolationScope());
```

Comment:
to sanity check: `runWithAsyncContext` forks the hub for the new async local storage and hence isolation and current scope, correct?

Reply:
Technically the answer to your question is no. `runWithAsyncContext` only forks the hub when there is an async context strategy set. In the browser this is not the case so the isolation scope will always be the same. This may seem odd, but as a side-effect will prevent users, who think async context keeping works in the browser, from running into race conditions.

(Side note: This is also the reason why I didn't add any forking tests in the core package, simply because we cannot make this assumption.)

Reply:
Right sorry, it's the async context strategy that takes care of it.